sycl/source/backend/opencl.cpp

```diff
@@ -0,0 +1,104 @@
+//==------- opencl.cpp - SYCL OpenCL backend -------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <detail/platform_impl.hpp>
+#include <detail/plugin.hpp>
+#include <detail/program_impl.hpp>
+#include <detail/queue_impl.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace opencl {
+using namespace detail;
+
+// Get the OpenCL plugin.
+static const plugin &getPlugin() {
+  static const plugin *OCLPlugin = nullptr;
+  if (OCLPlugin)
+    return *OCLPlugin;
+
+  const vector_class<plugin> &Plugins = pi::initialize();
+  for (const auto &Plugin : Plugins)
+    if (Plugin.getBackend() == backend::opencl) {
+      OCLPlugin = &Plugin;
+      break;
```

Comment:
The found plugin can be returned here. This eliminates the need for the following check and return.

Reply:
OK